File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        
        int i=0;
        int j=0;
        int n=str1.size();
        int m=str2.size();

        if(m>n) return false;

        while(i<n)
        {
            if(str1[i]==str2[j] || ( str1[i]=='z' ? 'a' : str1[i]+1  == str2[j]))
            {
                j++;
                if(j==m) return true;

            }

            i++;

        }


        return false;
        
    }
};

File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int i = 0, j = 0;
        int n = str1.length(), m = str2.length();

        if (m > n) return false;

        while (i < n) {
            char current = str1.charAt(i);
            char next = (current == 'z') ? 'a' : (char)(current + 1);
            if (current == str2.charAt(j) || next == str2.charAt(j)) {
                j++;
                if (j == m) return true;
            }
            i++;
        }

        return false;
    }
}